VisualRiders.PointOfSale.Project/Services/DiscountsService.cs

```diff
@@ -0,0 +1,195 @@
+
+        var discount = _repository.GetById(id);
+
+        if (discount == null) return null;
+
+        discountItem.Discount = discount;
+
+
+        if (dto.ProductId != null)
+        {
+            var product = _productsRepository.GetById(dto.ProductId.Value);
+            if (product == null)
+            {
+                throw new UnprocessableEntity($"Product with Id = {dto.ProductId.Value} does not exist");
+            }
+
+            discountItem.Product = product;
+        }
+
+        if (dto.ServiceId != null)
+        {
+            var service = _servicesRepository.GetById(dto.ServiceId.Value);
+            if (service == null)
+            {
+                throw new UnprocessableEntity($"Product with Id = {dto.ServiceId.Value} does not exist");
+            }
+
+            discountItem.Service = service;
+        }
+
+        if(discount.DiscountItems == null)
+        {
+            discount.DiscountItems = new List<DiscountItem>();
+        }
+        discount.DiscountItems.Add(discountItem);
+
+        _discountItemsRepository.Add(discountItem);
+        _discountItemsRepository.SaveChanges();
+
+        _repository.SaveChanges();
+
+        return _mapper.Map<ReadDiscountItemDto>(discountItem);
+    }
+
+    public ReadDiscountItemDto? UpdateItemById(int discountId, int itemId, CreateUpdateDiscountItemDto dto)
+    {
+        var discount = _repository.GetById(discountId);
+
+        if (discount == null) return null;
+
+        var discountItem = _discountItemsRepository.GetById(itemId);
+
+        if (discountItem == null)
+        {
+            throw new UnprocessableEntity($"Discount item with Id = {itemId} does not exist");
+        }
+
+        _mapper.Map(dto, discountItem);
+
+        if (dto.ProductId != null)
+        {
+            var product = _productsRepository.GetById(dto.ProductId.Value);
+            if (product == null)
+            {
+                throw new UnprocessableEntity($"Product with Id = {dto.ProductId.Value} does not exist");
+            }
+
+            discountItem.Product = product;
+        }
+
+        if (dto.ServiceId != null)
+        {
+            var service = _servicesRepository.GetById(dto.ServiceId.Value);
+            if (service == null)
+            {
+                throw new UnprocessableEntity($"Product with Id = {dto.ServiceId.Value} does not exist");
+            }
+
+            discountItem.Service = service;
+        }
+
+        _discountItemsRepository.SaveChanges();
+        _repository.SaveChanges();
+
+        return _mapper.Map<ReadDiscountItemDto>(discountItem);
+
+    }
+
+    public bool RemoveItemById(int discountId, int itemId)
+    {
+        var discount = _repository.GetById(discountId);
+
+        if (discount == null) return false;
+
+        var discountItem = _discountItemsRepository.GetById(itemId);
+
+        if(discountItem == null) return false;
+
+        if(discount.DiscountItems != null)
```

Comment:
Again, this is actually always `null` and to me, it seems that the item would never be removed as this `if` statement is always `false`. It would be enough to just remove the item in the `DiscountItemsRepository`.